flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -209,6 +213,7 @@ class SubtaskCheckpointCoordinatorImpl implements SubtaskCheckpointCoordinator {
         this.prepareInputSnapshot = prepareInputSnapshot;
         this.abortedCheckpointIds =
                 createAbortedCheckpointSetWithLimitSize(maxRecordAbortedCheckpoints);
+        this.maxRecordAbortedCheckpoints = maxRecordAbortedCheckpoints;
```

Comment:
Thanks for your review.

Actually, I have a question: why need the `SubtaskCheckpointCoordinatorImpl#DEFAULT_MAX_RECORD_ABORTED_CHECKPOINTS`? Why don't use the `max concurrent checkpoints` here?

Reply:
> Could we always use the max concurrent checkpoints and remove the SubtaskCheckpointCoordinatorImpl#DEFAULT_MAX_RECORD_ABORTED_CHECKPOINTS?

I think so.

> why need the SubtaskCheckpointCoordinatorImpl#DEFAULT_MAX_RECORD_ABORTED_CHECKPOINTS? Why don't use the max concurrent checkpoints here?

Probably because when it was being implemented by Myasuka (Yun Tang) and reviewed by me 2 years ago, we haven't figured out this trick with using `max concurrent checkpoints` :(

Reply:
If so, I can fix it. I should fix it in this pr and using a new commit, right?

Reply:
👍